### ISSUE TITLE
`SystemError` with `_string.formatter_field_name_split`

### ISSUE THREAD

**Issue by devdanzin** [pypy/pypy#5111]
It's possible to cause a `SystemError` by calling `_string.formatter_field_name_split` with an invalid value:

```python
import _string
_string.formatter_field_name_split(b'\xe1')
RPython traceback:
  File "pypy_interpreter.c", line 51609, in BuiltinCode1_fastcall_1
  File "pypy_module__string.c", line 449, in formatter_field_name_split
  File "pypy_objspace_std_10.c", line 13805, in TemplateFormatter_formatter_field_name_split
Traceback (most recent call last):
  File "<python-input-5>", line 1, in <module>
    _string.formatter_field_name_split(b'\xe1')
SystemError: unexpected internal exception (please report a bug): <CheckError object at 0x7b8708144ea0>; internal traceback was dumped to stderr
```

```
Python 3.10.14 (39dc8d3c85a7, Aug 27 2024, 14:32:27)
[PyPy 7.3.17 with GCC 10.2.1 20210130 (Red Hat 10.2.1-11)] on linux
```

Found using fusil by @vstinner.

**Comment by devdanzin** [comment]
Also happens for `_string.formatter_parser`:

```python
>>>> import _string
>>>> _string.formatter_parser(b"\xa1\xf2")
RPython traceback:
  File "pypy_interpreter.c", line 51609, in BuiltinCode1_fastcall_1
  File "pypy_objspace_std_8.c", line 41106, in TemplateFormatter_formatter_parser
Traceback (most recent call last):
  File "<python-input-4>", line 1, in <module>
    _string.formatter_parser(b"\xa1\xf2")
SystemError: unexpected internal exception (please report a bug): <CheckError object at 0x7936c1b2cf20>; internal traceback was dumped to stderr
```

**Comment by mattip** [comment]
Fixed in a608a540070, now raises an error. thanks for the report.
